File: 1037/main.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main(void) {
	int count;
	cin >> count;

	vector<int> v ;

	for (int i = 0 ; i < count ; i++ ) {
		int temp ;
		cin >> temp;
		v.push_back(temp);
	}

	sort(v.begin(), v.end());

	long long ans = (long long)v[0] * (long long) v[count-1];

	cout << ans;
}

File: 1037/Main.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.Collections;

public class Main {
	public static void main( String [] args ) {
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();

		ArrayList arr = new ArrayList<Integer>();

		for (int i = 0 ; i < N ; i++ ) {
			int temp = sc.nextInt();
			arr.add(temp);
		}

		Collections.sort(arr);

		long ans = (long)arr.get(0) * (long)arr.get(N-1);

		System.out.print(ans);
	}
}
		
